open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataEventProducer.java

```diff
@@ -97,58 +96,97 @@ private void publishEvent(String event) throws ConnectorCheckedException
-                                                                error.getMessage()),
-                                                                null,
-                                                                auditCode.getSystemAction(),
-                                                                auditCode.getUserAction());
-        }
-        catch (WakeupException error)
-        {
-            log.error("Wake up for shut down " + error.toString());
-        }
-        catch (Throwable error)
-        {
-            log.error("Exception in sendEvent " + error.toString());
-            KafkaOpenMetadataTopicConnectorErrorCode errorCode = KafkaOpenMetadataTopicConnectorErrorCode.ERROR_SENDING_EVENT;
-            String errorMessage = errorCode.getErrorMessageId() + errorCode.getFormattedErrorMessage(error.getClass().getName(),
-                                                                                                     topicName,
-                                                                                                     error.getMessage());
-
-            throw new ConnectorCheckedException(errorCode.getHTTPErrorCode(),
-                                                this.getClass().getName(),
-                                                methodName,
-                                                errorMessage,
-                                                errorCode.getSystemAction(),
-                                                errorCode.getUserAction(),
-                                                error);
-        }
-        finally
+        boolean                  eventSent = false;
+        long                     eventRetryCount = 0;
+
+        while (!eventSent)
         {
-            /*
-             * Producers have a thread and an in memory buffer
-             */
-            producer.flush();
+            try
+            {
+                log.debug("Sending message {0}" + event);
+                ProducerRecord<String, String> record = new ProducerRecord<>(topicName, localServerId, event);
+                producer.send(record).get();
+                eventSent = true;
+                messageSendCount++;
+            }
+            catch (ExecutionException error)
+            {
+                /*
+                 * This may be a simple timeout or something else more
+                 */
+                log.debug("Kafka had trouble sending event: " + event + "exception message is " + error.getMessage());
+
+                if( !isExceptionRetryable(error)) {
+                    /* kafka thinks this isn't a retryable problem */
+                    /* so let the caller try */
+                    log.error("Exception in sendEvent " + error.toString());
+                    KafkaOpenMetadataTopicConnectorErrorCode errorCode = KafkaOpenMetadataTopicConnectorErrorCode.ERROR_SENDING_EVENT;
```

Comment:
I notice the text for this event says the message isn't sent. True in terms of this connector's responsibility, but if queueing occurs further up the stack the message is confusing to the audit log reviewer

Reply:
Good to stick with this for our immediate kafka fix - can consider revisiting with a new connector implementation